api/v1alpha1/nicfirmwaretemplate_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+2025 NVIDIA CORPORATION & AFFILIATES
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// FirmwareTemplateSpec specifies a FW update policy for a given FW source ref
+type FirmwareTemplateSpec struct {
+	// NicFirmwareSourceRef refers to existing NicFirmwareSource CR on where to get the FW from
+	// +required
+	NicFirmwareSourceRef string `json:"nicFirmwareSourceRef"`
+	// UpdatePolicy indicates whether the operator needs to validate installed FW or upgrade it
+	// +kubebuilder:validation:Enum=Validate;Update
+	// +required
+	UpdatePolicy string `json:"updatePolicy"`
+}
+
+// NicFirmwareTemplateSpec defines the FW templates and node/nic selectors for it
+type NicFirmwareTemplateSpec struct {
+	// NodeSelector contains labels required on the node
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Maybe use:
`NodeSelector *corev1.NodeSelector json:"nodeSelector,omitempty" `

Reply:
I see here that we use the same map[string]string in the other template.

https://github.com/Mellanox/nic-configuration-operator/blob/main/api/v1alpha1/nicconfigurationtemplate_types.go#L93

Let's keep the same

Reply:
let's document behavior for an empty node selector as a part of the docstring.

Reply:
Done